test/app.test.js

```diff
@@ -1009,27 +1009,129 @@ describe('app', function() {
-      var UserAccount = PersistedModel.extend(
-        'UserAccount',
-        {name: String},
-        {
-          remoting: {normalizeHttpPath: true},
+    it.onServer('should honour `Model.remoting.normalizeHttpPath = false`', function(done) {
+      setupUserModels(app, {
+        remoting: {normalizeHttpPath: false},
+      }, function(err) {
+        if (err) return done(err);
+        request(app).get('/UserAccounts').expect(200, function(err) {
+          if (err) return done(err);
+          request(app).get('/UserAccounts/1/userRoles').expect(200, function(err) {
+            if (err) return done(err);
+            done();
+          });
         });
-      app.model(UserAccount);
-      UserAccount.attachTo(db);
+      });
+    });
 
-      app.use(loopback.rest());
-      request(app).get('/user-accounts').expect(200, done);
+    it.onServer('should honour `Model.remoting.normalizeHttpPath = true`', function(done) {
+      setupUserModels(app, {
+        remoting: {normalizeHttpPath: true},
+      }, function(err) {
+        if (err) return done(err);
+        request(app).get('/user-accounts').expect(200, function(err) {
+          if (err) return done(err);
+          request(app).get('/user-accounts/1/user-roles').expect(200, function(err) {
+            if (err) return done(err);
+            done();
+          });
+        });
+      });
     });
   });
+  describe('`app.remoting.rest.normalizeHttpPath` option', function() {
+    var app;
+    beforeEach(function() {
+      app = loopback();
+    });
+
+    it.onServer('should honour `remoting.rest.normalizeHttpPath = false`', function(done) {
+      app.set('remoting', {rest: {normalizeHttpPath: false}});
+      setupUserModels(app, null, function(err) {
+        if (err) return done(err);
+        request(app).get('/UserAccounts').expect(200, function(err) {
+          if (err) return done(err);
+          request(app).get('/UserAccounts/1/userRoles').expect(200, function(err) {
+            if (err) return done(err);
+            done();
+          });
+        });
+      });
+    });
+
+    it.onServer('should honour `remoting.rest.normalizeHttpPath = true`', function(done) {
+      app.set('remoting', {rest: {normalizeHttpPath: true}});
+      setupUserModels(app, null, function(err) {
+        if (err) return done(err);
+        request(app).get('/user-accounts').expect(200, function(err) {
+          if (err) return done(err);
+          request(app).get('/user-accounts/1/user-roles').expect(200, function(err) {
+            if (err) return done(err);
+            done();
+          });
+        });
+      });
+    });
+  });
+
+  describe(
+    '`Model.remoting.normalizeHttpPath` and `app.remoting.rest.normalizeHttpPath` options',
+    function() {
+      var app;
+      beforeEach(function() {
+        app = loopback();
+      });
+
+      it.onServer(
+        'should give `Model.remoting.normalizeHttpPath` precedence over ' +
```

Comment:
We are trying to avoid `should` prefix in new test names, see http://loopback.io/doc/en/contrib/style-guide.html. This applies to other test names added by this pull request too.

I am proposing to shorten the test name too, e.g. `prioritizes model-level setting over the app-level one`.

The full test name will include the title from `describe`, which already contains `normalizeHttpPath`. No need to repeat it here again...

```
...Model-level and app-level normalizeHttpPath option prioritizes model-level setting over the app-level one
```